### PR TITLE
build: Make Docker image extensible

### PR DESCRIPTION
**Pull request by sinhashubham95**
### SUMMARY
This change makes the docker image extensible, by adding the frontend files that can be used and modified.

### TESTING INSTRUCTIONS
`docker build -t superset -f Dockerfile`

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -77,6 +77,7 @@ RUN cd /app \
     && pip install --no-cache -r requirements/local.txt
 
 COPY --from=superset-node /app/superset/static/assets /app/superset/static/assets
+COPY --from=superset-node /app/superset-frontend /app/superset-frontend
```
Comment (zhaoyongjie) [inline Dockerfile:80]: The huge directory `node_modules` is in `superset-frontend` so I think it's not a good idea to copy this one.

**Comment by sinhashubham95** [comment]
@zhaoyongjie any update?

**Comment by EugeneTorap** [comment]
Hi @sinhashubham95
Can you explain why do we need the frontend source code in the final production image?
How do you plan to use & modify it in the final image?
If you know we build all UI source code into minified bundles and serve it via python server.
For us, two criteria for the docker image are important: the speed of the build and the lightness of the image.
Your PR makes the final image fatter and removing the node modules slows down the speed in Node section.

**Comment by EugeneTorap** [comment]
@villebro @rusackas Are you agree with me?

**Comment by rusackas** [comment]
I'm also curious as to the motivations and use cases here. If you can explain the workflow(s) that this enables, we might better be able to assess the approach you're proposing here.

**Comment by EugeneTorap** [comment]
@rusackas I guess the PR is better to close.

**Comment by rusackas** [comment]
Agreed. It's been sitting idle without any updates/rationale, so I'll just close it for the time being. @sinhashubham95 can certainly re-open and add context whenever it's convenient. The effort is appreciated in any case!

**Pull request closed without merging**